File: benchmarks/python/benchmark_logger.py
import os
import tempfile

import pytest
from logging_toolkit.implementation.logger import Logger, RotatingFileHandler


def test_logger_speed(benchmark):
    logger = Logger()
    handler = RotatingFileHandler(os.devnull)
    logger.add_handler(handler)
    benchmark(logger.info, "test message 42")


def test_logger_with_format(benchmark):
    logger = Logger(format_template="{timestamp} {level} {message}")
    handler = RotatingFileHandler(os.devnull)
    logger.add_handler(handler)
    benchmark(logger.info, "formatted message")


def test_logger_multiple_handlers(benchmark):
    logger = Logger()
    for _ in range(3):
        logger.add_handler(RotatingFileHandler(os.devnull))
    benchmark(logger.info, "multi handler message")


def test_logger_filtered(benchmark):
    from logging_toolkit.implementation.logger import LogFilter, LogLevel

    logger = Logger(filter=LogFilter(min_level=LogLevel.ERROR))
    handler = RotatingFileHandler(os.devnull)
    logger.add_handler(handler)
    benchmark(logger.info, "this should be filtered out")

File: benchmarks/java/BenchmarkLogger.java
package benchmarks.java;

import com.qe.test.logger.LogLevel;
import com.qe.test.logger.Logger;

public class BenchmarkLogger {

    public static BenchmarkResult run(int iterations) {
        Logger logger = new Logger("benchmark");

        long start = System.nanoTime();
        for (int i = 0; i < iterations; i++) {
            logger.log(LogLevel.INFO, "benchmark message " + i);
        }
        long end = System.nanoTime();

        long totalNanos = end - start;
        double opsPerSec = (double) iterations / (totalNanos / 1_000_000_000.0);
        double avgLatencyNanos = (double) totalNanos / iterations;

        return new BenchmarkResult("Logger.info", iterations, opsPerSec, avgLatencyNanos, totalNanos);
    }

    public static BenchmarkResult runWithFilter(int iterations) {
        Logger logger = new Logger("benchmark");
        logger.setThreshold(LogLevel.ERROR);

        long start = System.nanoTime();
        for (int i = 0; i < iterations; i++) {
            logger.log(LogLevel.INFO, "filtered message " + i);
        }
        long end = System.nanoTime();

        long totalNanos = end - start;
        double opsPerSec = (double) iterations / (totalNanos / 1_000_000_000.0);
        double avgLatencyNanos = (double) totalNanos / iterations;

        return new BenchmarkResult("Logger.info (filtered)", iterations, opsPerSec, avgLatencyNanos, totalNanos);
    }
}
